### ISSUE TITLE
Ordering

### ISSUE THREAD

**Issue by alexdavid** [Originate/morula#40]
Being able to define a run order in morula.yml (or dependencies) would be useful.

My use case is as follows:
I have sub-projects `A`, `B`, and `C`.  Building subproject `A` requires subproject `B` to have already been build because the output of building `B` is used to build `A`.

**Comment by kevgo** [comment]
There is a "before" and "after" key that you could use. Put project A into "before" and project "C" into after. Also keep in mind Morula runs projects in alphabetic order.

**Comment by kevgo** [comment]
We could add a key to the configuration called `dependencies`:

```yml
dependencies:
  B: A
```

Then Morula makes sure that the command is run in B before it is run in A.

How would this work for `morula changed`? What if B has changes, but A doesn't? Now Morula needs to build B. But in order to do that it also needs to build A. But it wouldn't build A, because A doesn't have changes. Does that mean it now also needs to build A, even though A doesn't have changes?

**Comment by alexdavid** [comment]
Yeah, ideally if we declare dependencies, if A changes then we need to build B as well. Although we don't need to run tests or deploy for B. Maybe we can declare commands in morula.yml? Although we should be careful to not turn it into another make replacement

**Comment by kevgo** [comment]
What about this:
* building with dependencies: `morula changed --include-dependencies build`
* testing without dependencies: `morula changed test`

**Comment by alexdavid** [comment]
Yeah I like that
